src/nvim/indent_c.c

```diff
@@ -508,29 +505,25 @@ get_indent_nolabel (     /* XXX */
  *   label:	if (asdf && asdfasdf)
  *		^
  */
-static int skip_label(linenr_T lnum, char_u **pp)
-{
-  char_u      *l;
-  int amount;
-  pos_T cursor_save;
+static int get_indent_after_label_pp(linenr_T lnum, char_u **pp)
+  FUNC_ATTR_NONNULL_ALL
+{ /* XXX */
+  char_u *l = ml_get(lnum);
+  int indent;
 
-  cursor_save = curwin->w_cursor;
-  curwin->w_cursor.lnum = lnum;
-  l = get_cursor_line_ptr();
-  /* XXX */
+  // We repeatedly call "ml_get(lnum)" because every time it's called (or any
+  // time we call a function that calls it, marked with an 'XXX' comment) the
+  // returned pointer is invalidated.
```

Comment:
If we update the function comment to the new format (and replace tabs with spaces), it could be helpful to have a note or warning about this. Perhaps something like
`/// @warning This function invalidates pointers obtained with the ml_get_*() functions`.